### ISSUE TITLE
ZeroDivisionError: float division by zero when use apply_redactions()

### ISSUE THREAD

**Issue by burke-up** [pymupdf/PyMuPDF#2943]
### Description of the bug
```
Traceback (most recent call last):
  File "D:\project\mytest.py", line 14, in <module>
    page.apply_redactions()
  File "D:\project\venv\lib\site-packages\fitz\utils.py", line 4261, in apply_redactions
    trect = center_rect(annot_rect, text, fname, fsize)
  File "D:\project\venv\lib\site-packages\fitz\utils.py", line 4220, in center_rect
    h = math.ceil(text_width / limit) * line_height  # estimate rect height
ZeroDivisionError: float division by zero
```

### How to reproduce the bug

first put the file Equity Compensation Master.pdf [Equity Compensation Master.pdf](https://github.com/pymupdf/PyMuPDF/files/13768000/Equity.Compensation.Master.pdf)

 to data/input/Equity Compensation Master.pdf
then run the code:
```
import fitz
import json

doc = fitz.open(filename="data/input/Equity Compensation Master.pdf")

page = doc[0]
info = json.loads(page.get_text("json"))
span = info.get("blocks")[3].get("lines")[0].get("spans")[0]

span_text = span.get("text","")
print("span: bbox=",span["bbox"]," text=",span_text)
page.add_redact_annot(span["bbox"], text="New Text", fontsize=span["size"], fill=(1, 1, 0))
page.apply_redactions()
 ```

### PyMuPDF version

Built from source

### Operating system

Windows

### Python version

3.10

**Comment by JorjMcKie** [comment]
Cannot reproduce:
```python
>py -3.10 test.py


PyMuPDF 1.23.8: Python bindings for the MuPDF 1.23.7 library.
Version date: 2023-12-19 00:00:01.
Built for Python 3.10 on win32 (64-bit).

span: bbox= [272.4049987792969, 309.27484130859375, 447.60748291015625, 343.45452880859375]  text= Frank Demmler
```

BTW: any particular reason why you would use the JSON output instead of the original "dict"?

**Comment by JorjMcKie** [comment]
Further investigation revealed that there is an error in the rebased implementation only, which in turn only occurs if the rectangle in `page.add_redact_annot(rect, ...)` is not a `fitz.Rect` but only a rect-like (4-float sequence).
So in your case, you can avoid this by always using `page.add_redact_annot(fitz.Rect(span["bbox"]), ...)`.

**Comment by burke-up** [comment]
> Further investigation revealed that there is an error in the rebased implementation only, which in turn only occurs if the rectangle in `page.add_redact_annot(rect, ...)` is not a `fitz.Rect` but only a rect-like (4-float sequence). So in your case, you can avoid this by always using `page.add_redact_annot(fitz.Rect(span["bbox"]), ...)`.

it works  when change code to page.add_redact_annot(fitz.Rect(span["bbox"]), ...).

**Comment by JorjMcKie** [comment]
Thanks for your feedback and I am glad it works for you!
Our internal rules however require that keep an issue open until the fixed version is available on PyPI.
So I am going to reopen this.

**Comment by julian-smith-artifex-com** [comment]
This is fixed in release candidate 1.23.9rc1; see: #2960
